packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -440,6 +442,65 @@ class DirectStripeRoutes {
 
     return {};
   }
+
+  /**
+   * Get a list of subscriptions with a FxA UID and email address.
```

Comment:
We have existing API routes for [`listActive`](https://github.com/mozilla/fxa/blob/9589fc7906810c085ce755852f42804476dca8e5/packages/fxa-auth-server/lib/routes/subscriptions.js#L345) and [`getCustomer`](https://github.com/mozilla/fxa/blob/9589fc7906810c085ce755852f42804476dca8e5/packages/fxa-auth-server/lib/routes/subscriptions.js#L371) - do neither of these work for this purpose? 

The `listActive` route is kind of "legacy" since it was the view of the FxA DB table, made obsolete by moving to direct Stripe access but still used on payments-server. The customer data should list subscriptions, though

Reply:
Alternatively, if we really *do* need this new API route - I'd say maybe drop the use of `subscriptionsToResponse` on `stripeHelper` and return the Stripe data directly. I think `subscriptionsToResponse` is a kind of legacy adapter to make data look like subhub responses, but we can move away from that. (@bbangert could probably speak to that better, e.g. maybe we need to do some sanitizing for privacy / security on this data as it passes by)

Reply:
D'oh.... `fetchCustomer` isn't what I needed here; I'll change it to `customer`.  Thanks!

As for the new route, I landed on that (after some struggling/indecisiveness) because:
 - Hapi doesn't offer a way to do multiple auth strategies if both of the strategies use the same header scheme (the `authorization` header in this case); we would need to write a single strategy that check both and auth on the first valid token.
 - Since the requests don't look the same, we'd need to move the `uid` and `email` query param validation somewhere, instead of declaring them with the route.

If there's a clean way to have a single route in Hapi, I'd love to know.